src/test/java/software/amazon/awssdk/crt/test/Mqtt5ClientTest.java

```diff
@@ -218,77 +218,81 @@ public void New_UC2() {
+                .withSessionExpiryIntervalSeconds(1000L)
+                .withUsername(mqtt5BasicAuthUsername)
+                .withWill(willPacketBuilder.build())
+                .withWillDelayIntervalSeconds(1000L);
+
+                ArrayList<UserProperty> userProperties = new ArrayList<UserProperty>();
+                userProperties.add(new UserProperty("Hello", "World"));
+                connectBuilder.withUserProperties(userProperties);
+
+                Mqtt5ClientOptionsBuilder builder = new Mqtt5ClientOptionsBuilder(mqtt5DirectMqttHost, mqtt5DirectMqttPort);
+                builder.withBootstrap(bootstrap)
+                .withConnackTimeoutMs(100L)
+                .withConnectOptions(connectBuilder.build())
+                .withExtendedValidationAndFlowControlOptions(ExtendedValidationAndFlowControlOptions.NONE)
+                .withHostName(mqtt5DirectMqttHost)
+                .withLifecycleEvents(new LifecycleEvents() {
+                    @Override
+                    public void onAttemptingConnect(Mqtt5Client client, OnAttemptingConnectReturn onAttemptingConnectReturn) {}
+
+                    @Override
+                    public void onConnectionSuccess(Mqtt5Client client, OnConnectionSuccessReturn onConnectionSuccessReturn) {}
+
+                    @Override
+                    public void onConnectionFailure(Mqtt5Client client, OnConnectionFailureReturn onConnectionFailureReturn) {}
+
+                    @Override
+                    public void onDisconnection(Mqtt5Client client, OnDisconnectionReturn onDisconnectionReturn) {}
+
+                    @Override
+                    public void onStopped(Mqtt5Client client, OnStoppedReturn onStoppedReturn) {}
+                })
+                .withMaxReconnectDelayMs(1000L)
+                .withMinConnectedTimeToResetReconnectDelayMs(1000L)
+                .withMinReconnectDelayMs(1000L)
+                .withOfflineQueueBehavior(ClientOfflineQueueBehavior.FAIL_ALL_ON_DISCONNECT)
+                .withAckTimeoutSeconds(1000L)
+                .withPingTimeoutMs(1000L)
+                .withPort(mqtt5DirectMqttPort)
+                .withPublishEvents(new PublishEvents() {
+                    @Override
+                    public void onMessageReceived(Mqtt5Client client, PublishReturn publishReturn) {}
+                })
+                .withRetryJitterMode(JitterMode.Default)
+                .withSessionBehavior(ClientSessionBehavior.CLEAN)
+                .withSocketOptions(socketOptions);
+                // Skip websocket, proxy options, and TLS options - those are all different tests
+
+                try (Mqtt5Client client = new Mqtt5Client(builder.build())) {
+                    assertNotNull(client);
+                }
 
-            elg.close();
-            hr.close();
-            bootstrap.close();
-            socketOptions.close();
-            client.close();
+                elg.close();
```

Comment:
These shouldn't be necessary.  If done correctly you should only ever have to manually close something that you let escape the try block via doing an addRef on it

Reply:
Good catch - I missed that one! I did a check and now anywhere `close()` is called is either in commented out tests (which I need to get back to and/or fix) and in the overflow tests.